waspc/src/Wasp/JsImport.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+
+module Wasp.JsImport
+  ( JsImport (..),
+    JsImportName (..),
+    JsImportData,
+    JsImportAlias,
+    JsImportPath,
+    getJsImportData,
+  )
+where
+
+import Data.Data (Data)
+import Data.List (isPrefixOf)
+import StrongPath (Dir', File', Path, Posix, Rel)
+import qualified StrongPath as SP
+
+data JsImport = JsImport
+  { -- | What is being imported.
+    _name :: JsImportName,
+    -- | Path from which we are importing.
+    _path :: JsImportPath,
+    -- | Alias for the imported name
+    _importAlias :: Maybe JsImportAlias
+  }
+  deriving (Show, Eq, Data)
+
+type JsImportPath = Path Posix (Rel Dir') File'
+
+type Identifier = String
+
+type JsImportAlias = String
+
+data JsImportName
+  = -- | Represents external imports like @import Identifier from "file.js"@
+    JsImportModule Identifier
+  | -- | Represents external imports like @import { Identifier } from "file.js"@
+    JsImportField Identifier
+  deriving (Show, Eq, Data)
+
+type JsImportIdentifier = String
+
+type JsImportWhat = String
+
+type JsImportStatement = String
+
+type JsImportData = (JsImportIdentifier, JsImportStatement)
+
+getJsImportData :: JsImportPath -> JsImportName -> Maybe JsImportAlias -> JsImportData
```

Comment:
I am just somehow tempted to take this one, and two functions below: getImportStatement and getFirstPartOfJsImport and put them all into one function with stuff in `where`, because having this guys standalone is a bit weird. But ok, we can leave it like this (I know you copied this from before, I wrote it like this before).